src/python/grpcio/grpc/experimental/aio/_interceptor.py

```diff
@@ -0,0 +1,334 @@
+        return await (await self._interceptors_task).trailing_metadata()
+
+    async def code(self) -> grpc.StatusCode:
+        return await (await self._interceptors_task).code()
+
+    async def details(self) -> str:
+        return await (await self._interceptors_task).details()
+
+    async def debug_error_string(self) -> Optional[str]:
+        return await (await self._interceptors_task).debug_error_string()
+
+    def __await__(self):
+        call = yield from self._interceptors_task.__await__()
+        response = yield from call.__await__()
+        return response
+
+
+class UnaryUnaryCallRpcError(_base_call.UnaryUnaryCall):
+    """Final UnaryUnaryCall class finished with an RpcError."""
+    _error: grpc.RpcError
+
+    def __init__(self, error: grpc.RpcError) -> None:
+        self._error = error
+
+    def cancel(self) -> bool:
+        return False
+
+    def cancelled(self) -> bool:
+        return False
+
+    def done(self) -> bool:
+        return True
+
+    def add_done_callback(self, unused_callback) -> None:
+        raise NotImplementedError()
+
+    def time_remaining(self) -> Optional[float]:
+        raise NotImplementedError()
+
+    async def initial_metadata(self) -> Optional[MetadataType]:
+        return None
+
+    async def trailing_metadata(self) -> Optional[MetadataType]:
+        return self._error.initial_metadata()
+
+    async def code(self) -> grpc.StatusCode:
+        return self._error.code()
+
+    async def details(self) -> str:
+        return self._error.details()
+
+    async def debug_error_string(self) -> Optional[str]:
+        return self._error.debug_error_string()
+
+    def __await__(self):
+        raise self._error
+
+
+class UnaryUnaryCallResponse(_base_call.UnaryUnaryCall):
+    """Final UnaryUnaryCall class finished with a response."""
+    _response: ResponseType
+
+    def __init__(self, response: ResponseType) -> None:
+        self._response = response
+
+    def cancel(self) -> bool:
+        return False
+
+    def cancelled(self) -> bool:
+        return False
+
+    def done(self) -> bool:
+        return True
+
+    def add_done_callback(self, unused_callback) -> None:
+        raise NotImplementedError()
+
+    def time_remaining(self) -> Optional[float]:
+        raise NotImplementedError()
+
+    async def initial_metadata(self) -> Optional[MetadataType]:
+        return None
+
+    async def trailing_metadata(self) -> Optional[MetadataType]:
+        return None
+
+    async def code(self) -> grpc.StatusCode:
+        return grpc.StatusCode.OK
+
+    async def details(self) -> str:
+        return ''
+
+    async def debug_error_string(self) -> Optional[str]:
+        return None
+
+    def __await__(self):
+        if False:  # pylint: disable=W0125
+            # This code path is never used, but a yield statement is needed
+            # for telling the interpreterthat __await__ is a generator.
+            yield None
```

Comment:
nit: Can we add a TODO here to remove this trick? Potentially, we could make this class a generator. It doesn't block this PR.